### DISCUSSION TITLE
AnimateCC alternative publishing guide

### DISCUSSION THREAD

**Discussion by M4rkY4** [airsdk/Adobe-Runtime-Support#2713] [Guides & Tutorials]
Hi,

as many of you (and me as well) enjoy developing via Animate CC, I would like to point out that it's possible to use Animate in development, but build to iOS/Android "the right way", via Intellij IDEA with the help of APM and SDK Manager.

You fully set up Animate CC (ANEs/SWCs) to have no compile / autocomplete issues, run / debug your app with Animate ( [Ctrl] [Enter] ) which will auto-create SWF, and just use that SWF for building in Intellij IDEA (fully set up as well). Please remember to disable debug when exporting SWF for final builds.

The main trick is to put FLA in the root folder, SWF will be created in the root folder as well (so that it will not be erased when building, like everything inside /out folder), and then you set up to skip compilation and link to that SWF.

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/93158631/73f5cbe4-a1fb-4d7a-b427-0e97bbc5cfc9)

Surely, you will be unable to use Intellij IDEA stellar debugging process, but at least you can avoid outdated AnimateCC build capabilities. Additionally, with each build you will have this message, just click [Yes]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/93158631/1f97d40b-2906-41a6-bd68-f8dc4b349351)

Intellij IDEA has a Perpetual Fallback license, so you can buy in once with a year subscription price. Not in any way connected to them, just love this IDE :)

Maybe this method has some drawbacks, I just tested it with a simplest project with a single ANE, but it looks promising.

I hope this helps :)

PS: I have no idea if it works with VsCode, if you were able to do the same with it, sharing your experience might help others.

**Comment by 2jfw** [comment]
Not sure about using the debug SWF, since it is likely missing code-optimization, which is only done when creating a release build (or by adding "-optimize=true" compiler arg). Depending on what your application does this can cause noticeable performance issues... especially for games not recommended

**Reply by M4rkY4** [reply]
Thanks, fair point. As I understand, disabling debug/telemetry and "omitting trace statements" in Animate CC will not be enough, right? Maybe, creating desktop build just for SWF will do the trick?

**Reply by 2jfw** [reply]
Sorry, but "Enough" is too broad/ambiguous ;)
With the Debug SWFs there will be no code optimization and this can (but must not) heavily affect the performance of your app. You can simply benchmark with Adobe Scout and compare the release with debug version to find out how both perform.

I would recommend you to use Animate only to create SWCs for the purpose of embedding and use a real IntelliJ project for coding/debugging/creating release build etc.
I am pretty sure that creating a SWC from Animate will consider code optimization.

**Comment by Jason0247** [comment]
Sorry, I didn't quite understand what you were saying.
My workflow is as follows: when developing Windows applications, I first publish the AIR runtime package through Animate. Previously, in Animate's publishing settings, I set the path to the published application runtime package. When using Ctrl+enter, SWF will automatically overwrite the runtime package folder.
So if that's the case, will using Animate (ctrl+enter) to publish programs affect efficiency?

**Reply by 2jfw** [reply]
> So if that's the case, will using Animate (ctrl+enter) to publish programs affect efficiency?

Yes, it will negatively affect your application's performance - CTRL+Enter will create an SWF which contains debug information and which is not code-optimized. You should not distribute that SWF or make it a release version.
The performance impact depends on what your app is actually doing and you would need to benchmark the different outputs on your own to find out how much it slows down.

**Reply by Jason0247** [reply]
I have conducted a simple test on the issue of ctrl+enter.
When I keep using trace in Animate, it does become very slow, but when I start exe in the runtime package, it does not affect efficiency.
Compared to double clicking on exe to start the program in Animate, the speed is indeed much slower (but it seems that in the released version, when double clicking on exe to run, it is not affected by debugging. I wonder if the system automatically blocks it?)
thanks

**Comment by Jason0247** [comment]
I just used Adobe Scout CC for testing
Indeed, when I use ctrl+enter and double-click to open exe, Scout will prompt me that I am currently using the debug version. But when I use Aniname's publishing (more settings - publishing) Scout will not have a debug version prompt.
As for the unclear changes in efficiency, my program has been reading sequence diagrams and playing them, with image sizes of 3840 * 2160 pixels. DisplayList rendering takes up a lot of time (I have been trying to solve this problem because I saw HTML playing high-resolution images)

**Comment by M4rkY4** [comment]
> I just used Adobe Scout CC for testing Indeed, when I use ctrl+enter and double-click to open exe, Scout will prompt me that I am currently using the debug version. But when I use Aniname's publishing (more settings - publishing) Scout will not have a debug version prompt. As for the unclear changes in efficiency, my program has been reading sequence diagrams and playing them, with image sizes of 3840 * 2160 pixels. DisplayList rendering takes up a lot of time (I have been trying to solve this problem because I saw HTML playing high-resolution images)

Hi,
I believe, before production export you just need to disable debug in your SWF and export/publish (ctrl-enter should work as well, you can test that) your final swf

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/93158631/e802c913-3788-482d-8c15-0c68754b07ec)

In case you still see some performance drop, which I assume is unlikely, you can always export some build via Animate, open it as an archive and copy your optimized SWF.

It would be nice if you would be able to confirm or debunk these performance concerns. Thanks)

**Reply by Jason0247** [reply]
Yes, this operation is completely possible.
